File: Python-Stack/BankAccount.py
class BankAccount:
    def __init__(self, int_rate, balance):
        self.int_rate = int_rate
        self.balance = balance
    def deposit(self, amount):
        self.balance += amount
    def withdraw(self, amount):
        self.balance -= amount
    def display_account_info(self):
        print("BankAccount : "+ str(self.int_rate) +" | "+str(self.balance))
    def yield_interest(self):
        interest = self.balance / self.int_rate
        self.withdraw(interest)

account_one = BankAccount(10, 1000)
account_one.display_account_info()
account_one.yield_interest()
account_one.display_account_info()

account_two = BankAccount(12, 1000)
account_two.display_account_info()
account_two.yield_interest()
account_two.display_account_info()

File: Java-Stack/BankAccount/src/BankAccount.java
public class BankAccount {
    // MEMBER VARIABLES
    private double checkingBalance;
    private double savingsBalance;

    private static int accounts;
    private static double totalMoney;

    // CONSTRUCTOR
    // Be sure to increment the number of accounts
    public BankAccount(double checkingBalance, double savingsBalance){
        setCheckingBalance(checkingBalance);
        setSavingsBalance(savingsBalance);
        accounts++;
        totalMoney += checkingBalance+savingsBalance;
    }

    // GETTERS
    // for checking, savings, accounts, and totalMoney
    public double getCheckingBalance() {
        return checkingBalance;
    }

    private void setCheckingBalance(double checkingBalance) {
        this.checkingBalance = checkingBalance;
    }

    public double getSavingsBalance() {
        return savingsBalance;
    }

    private void setSavingsBalance(double savingsBalance) {
        this.savingsBalance = savingsBalance;
    }

    public static int getAccounts() {
        return accounts;
    }

    // METHODS
    // deposit
    // - users should be able to deposit money into their checking or savings account
    public void depositIntoChecking(double amount){
        this.checkingBalance += amount;
        totalMoney += amount;
    }
    public void depositIntoSaving(double amount){
        this.savingsBalance += amount;
        totalMoney += amount;
    }

    // withdraw
    // - users should be able to withdraw money from their checking or savings account
    // - do not allow them to withdraw money if there are insufficient funds
    // - all deposits and withdrawals should affect totalMoney
    public void withdrawFromChecking(double amount){
        if(this.checkingBalance < amount){
            System.out.println("You have No enough Money");
            return;
        }
        this.checkingBalance -= amount;
        totalMoney -= amount;
        System.out.println(amount+"$ سحبوا منك ه القد");
    }
    public void withdrawFromSaving(double amount){
        if(this.savingsBalance < amount){
            System.out.println("You have No enough Money");
            return;
        }
        this.savingsBalance -= amount;
        totalMoney -= amount;
        System.out.println(amount+"$ سحبوا منك ه القد");
    }


    // getBalance
    // - display total balance for checking and savings of a particular bank account

    public static double getTotalMoney(){
        return totalMoney;
    }
}
